src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -198,40 +199,24 @@ void Convolution<
       padHBottom);
 
   output.set_size(wConv * hConv * outSize, batchSize);
-  outputTemp = arma::Cube<eT>(output.memptr(), wConv, hConv,
-      outSize * batchSize, false, false);
+  //output.zeros();
+  outputTemp = arma::Cube<eT>(output.memptr(), wConv, hConv, outSize * batchSize, false, false);
   outputTemp.zeros();
 
-  for (size_t outMap = 0, outMapIdx = 0, batchCount = 0; outMap <
-      outSize * batchSize; outMap++)
+  arma::Cube<eT> inp = (padWLeft | padWRight | padHTop | padHBottom) != 0 ? inputPaddedTemp : inputTemp;
```

Comment:
We generally prefer names that are much more intuitive. Let me know what you think.

Reply:
I needed that variable earlier, don't anymore. Will remove it.